db/db_filesnapshot.cc

```diff
@@ -349,7 +382,13 @@ Status DBImpl::GetLiveFilesStorageInfo(
   TEST_SYNC_POINT("CheckpointImpl::CreateCheckpoint:SavedLiveFiles2");
 
   if (s.ok()) {
-    s = FlushWAL(false /* sync */);
+    // To maximize the effectiveness of track_and_verify_wals_in_manifest,
+    // sync WAL when it is enabled.
+    s = FlushWAL(
+        immutable_db_options_.track_and_verify_wals_in_manifest /* sync */);
```

Comment:
Surprisingly this can corrupt the DB. The `SyncWAL()` it calls when `track_and_verify_wals_in_manifest == true` will, at least when `manual_wal_flush == false`, look at `GetFileSize()` after syncing finishes to determine the synced size. But new writes could have come to that WAL in the meantime causing `GetFileSize()` to be higher than the actual synced size. Then a recovery following unsynced data loss can fail to open like this:

```
Error opening DB: Corruption: Size mismatch: WAL (log number: 440) in MANIFEST is 63351 bytes , but actually is 62772 bytes on disk
```

I'll try to see if moving the `GetFileSize()` earlier can solve it...